web3indexer/worker.py

```diff
@@ -34,12 +41,27 @@ def run(self):
                 # Special case the stop task.
                 if task is STOP_TASK:
                     return
-                executor.submit(
-                    self.collectors[task.collector].collect_with_retry,
-                    self.dispatcher,
-                    self.w3,
-                    task,
-                )
+                if isinstance(task, FetchBlockTask):
```

Comment:
i think it was intended in the original design that each task (e.g. FetchBlockTask, ProcessLogTask) specifies a collector (e.g. BlockFetcher, LogProcessor). the worker logic here will simply forward the task to the specified collector (as in line ~59 `executor.submit(self.collectors[task.collector].collect_with_retry, ...)`)